src/Server/OperationParams.php

```diff
@@ -84,57 +106,69 @@ public static function create(array $params, bool $readonly = false): OperationP
             'extensions' => null,
         ];
 
-        if ($params['variables'] === '') {
-            $params['variables'] = null;
-        }
-
-        // Some parameters could be provided as serialized JSON.
-        foreach (['extensions', 'variables'] as $param) {
-            if (! is_string($params[$param])) {
-                continue;
-            }
-
-            $tmp = json_decode($params[$param], true);
-            if (json_last_error() !== JSON_ERROR_NONE) {
-                continue;
+        array_walk($params, static function (&$value): void {
+            if ($value !== '') {
+                return;
             }
 
-            $params[$param] = $tmp;
-        }
+            $value = null;
+        });
 
         $instance->query      = $params['query'];
         $instance->queryId    = $params['queryid'] ?? $params['documentid'] ?? $params['id'];
         $instance->operation  = $params['operationname'];
-        $instance->variables  = $params['variables'];
-        $instance->extensions = $params['extensions'];
+        $instance->variables  = static::decodeIfJSON($params['variables']);
+        $instance->extensions = static::decodeIfJSON($params['extensions']);
         $instance->readOnly   = $readonly;
 
-        // Apollo server/client compatibility: look for the queryid in extensions
-        if (isset($instance->extensions['persistedQuery']['sha256Hash']) && strlen($instance->query ?? '') === 0 && strlen($instance->queryId ?? '') === 0) {
+        // Apollo server/client compatibility
+        if (
+            isset($instance->extensions['persistedQuery']['sha256Hash'])
+            && $instance->query === null
+            && $instance->queryId === null
+        ) {
             $instance->queryId = $instance->extensions['persistedQuery']['sha256Hash'];
         }
 
         return $instance;
     }
 
     /**
-     * @return mixed
+     * @deprecated
```

Comment:
This was not deprecated so we might need to add it to changelog? I'd also add some note on why deprecation happens.

Reply:
Given we are working towards a new major version anyways, I just removed them and put that into the changelog. I don't like getters, especially ones that prevent direct access to an array. It prevents so many useful things that can be done with arrays.